Sources/Instructions/Core/Internal/CoachMarksViewController.swift

```diff
@@ -27,7 +27,7 @@ import UIKit
 
 // MARK: - Main Class
 /// Handles a set of coach marks, and display them successively.
-class CoachMarksViewController: UIViewController {
+public class CoachMarksViewController: UIViewController {
```

Comment:
This class (and its fields) needs to remain `internal`.

Reply:
I will add an example for this. As i want to get currentCoachMarkView to add a animation when it will or did show.

Reply:
do you have some advice for custom animation of CoachMarkView. I think what you said is right, CoachMarksViewController has to remain internal

Reply:
Adding animations to coach marks has to be done in another PR.

I'm currently drafting how it will work and I plan to use delegate methods as usual. I would use something in the spirit of `UIViewControllerTransitionCoordinator`, which would be provided to developers through the delegate methods. The developer could then pass the animation block to its `animate` method.

It could look like this:

```swift
func coachMarksController(_ coachMarksController: CoachMarksController, willAnimateAppearanceOfCoachMark view: UIView, at index: Int, using coordinator: CoachMarkTransitionCoordinator) {
    coordinator.animate {
        view.alpha = 0.0
    }
}
```

That said, I plan to work on this right after fixing the few open issues about the layout engine, so you're probably better off focusing on other aspects of Instructions. I need to be able to animate coach marks as well for an app that I'm making. ;)